forte/solvers/active_space_solver.py

```diff
@@ -0,0 +1,155 @@
+from forte.core import flog
+
+from forte.solvers.solver import Feature, Solver
+
+from forte.forte import StateInfo
+from forte.forte import ForteOptions
+from forte import to_state_nroots_map
+from forte import forte_options
+from forte.forte import make_active_space_ints, make_active_space_solver
+
+
+class ActiveSpaceSolver(Solver):
+    """
+    A class to diagonalize the Hamiltonian in a subset of molecular orbitals.
+    """
+    def __init__(
+        self,
+        input,
+        states,
+        type,
+        active=None,
+        restricted_docc=None,
+        frozen_docc=None,
+        gas1=None,
+        gas2=None,
+        gas3=None,
+        gas4=None,
+        gas5=None,
+        gas6=None,
+        e_convergence=1.0e-10,
+        r_convergence=1.0e-6,
+        options=None,
+        cbh=None
+    ):
+        """
+        Initialize a FCI object
+
+        Parameters
+        ----------
+        input: Solver
+            The solver that will provide the molecular orbitals
+        states: dict()
+            A dictionary of StateInfo -> list(float) of all the states that will be computed.
```

Comment:
What does this list do? I suspect something with state-averaging.

Reply:
I added a long comment on this option.